src/components/shared/ReconfigurationDrawer.tsx

```diff
@@ -0,0 +1,162 @@
+import { t, Trans } from '@lingui/macro'
+import { Button, Drawer, DrawerProps, Input, Space } from 'antd'
+import { constants, utils } from 'ethers'
+import { NetworkContext } from 'contexts/networkContext'
+import { useContext, useState } from 'react'
+import { ThemeContext } from 'contexts/themeContext'
+import { drawerWidth } from 'utils/drawerWidth'
+
+import ReconfigurationOption from 'components/shared/ReconfigurationOption'
+
+import {
+  ballotStrategies,
+  customStrategy,
+  Strategy,
+} from 'constants/ballotStrategies/ballotStrategies'
+
+function CustomStrategyForm({
+  address,
+  setAddress,
+}: {
+  address?: string
+  setAddress: (address: string) => void
+}) {
+  const { signerNetwork } = useContext(NetworkContext)
+  return (
+    <div>
+      <Input
+        style={{ width: 400 }}
+        value={address}
+        placeholder={constants.AddressZero}
+        onChange={e => setAddress(e.target.value.toLowerCase())}
+      />
+      <p>
+        <Trans>
+          The address of any smart contract deployed on {signerNetwork} that
+          implements
+        </Trans>{' '}
+        <a
+          href="https://github.com/jbx-protocol/juice-contracts-v1/blob/05828d57e3a27580437fc258fe9041b2401fc044/contracts/FundingCycles.sol"
+          target="_blank"
+          rel="noopener noreferrer"
+        >
+          <Trans>this interface</Trans>
+        </a>
+        .
+      </p>
+    </div>
+  )
+}
+
+export default function ReconfigurationDrawer({
+  visible,
+  onClose,
+  initialSelectedStrategy,
+  onSave,
+}: {
+  visible: boolean
+  onClose: () => void
+  initialSelectedStrategy: Strategy
+  onSave: Function
+}) {
+  const { colors } = useContext(ThemeContext).theme
+
+  const [selectedStrategy, setSelectedStrategy] = useState<Strategy>(
+    initialSelectedStrategy,
+  )
+
+  const selectedStrategyIndex = ballotStrategies().findIndex(s => {
+    return s.address?.toLowerCase() === selectedStrategy.address?.toLowerCase()
+  })
+
+  const [customStrategyAddress, setCustomStrategyAddress] = useState<string>(
+    selectedStrategyIndex === -1 ? initialSelectedStrategy.address : '', // only set if selected strategy is custom
+  )
+
+  const drawerStyle: Partial<DrawerProps> = {
```

Comment:
nit pick, i think i'd rather still pass this in externally. the drawer itself doesn't have to care about how it's being styled